overlays/nodejs/serverless/function.ts

```diff
@@ -106,9 +129,10 @@ export class Function extends pulumi.ComponentResource {
 
     constructor(name: string,
         options: FunctionOptions,
-        func: Handler,
+        func: FuncSignature,
```

Comment:
This will, I'm nearly certain - lead to much worse intellisense - since TypeScript can't infer the signature for callers.  I'm not a huge fan of losing this.

Should we consider:
1. A different class that inherits from `aws.serverless.Function` that accepts a different argument?
2. Moving the function parameter inside the args bag and allowing either `func: ` or `factory: `?
3. Something else?